src/convertToObject.js

```diff
@@ -7,6 +7,16 @@
  */
 function convertToObject(sourceString) {
   // write your code here
+  if (sourceString) {
+    return sourceString
+      .split(';')
+      .map((row) => row.split(':').map((param) => param.trim()))
+      .filter((row) => row.length > 1)
+      .map((e) => ({ [e[0]]: e[1] }))
+      .reduce((prew = {}, e) => ({ ...prew, ...e }));
```

Comment:
The spread operator inside the `reduce` callback is not the most efficient way to merge objects because it creates a new object on each iteration. You can directly assign the new property to the accumulator to avoid unnecessary object copies.